File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/java/domain/sample/AbstractAddress.java
package com.coupang.springframework.data.requery.java.domain.sample;

import com.coupang.springframework.data.requery.domain.AbstractComponent;
import io.requery.Embedded;
import lombok.Getter;
import lombok.Setter;

/**
 * AbstractAddress
 *
 * @author debop@coupang.com
 * @since 18. 6. 14
 */
@Getter
@Setter
@Embedded
public class AbstractAddress extends AbstractComponent {

    private static final long serialVersionUID = 6292061477905475831L;

    protected String country;
    protected String city;
    protected String streetName;
    protected String streetNo;

}

File: src/test/kotlin/com/coupang/springframework/data/requery/domain/sample/AbstractAddress.kt
package com.coupang.springframework.data.requery.domain.sample

import com.coupang.kotlinx.objectx.ToStringBuilder
import com.coupang.springframework.data.requery.domain.AbstractComponent
import io.requery.Column
import io.requery.Embedded
import io.requery.Transient

/**
 * com.coupang.springframework.data.requery.domain.sample.AbstractAddress
 * @author debop
 * @since 18. 5. 30
 */
@Embedded
abstract class AbstractAddress: AbstractComponent() {

    @get:Column(length = 6, nullable = false)
    abstract var zipcode: String

    @get:Column(length = 32)
    abstract var country: String?

    @get:Column(length = 32)
    abstract var city: String?

    @get:Column(length = 128)
    abstract var streetName: String?

    @get:Column(length = 128)
    abstract var streetNo: String?

    @Transient
    override fun buildStringHelper(): ToStringBuilder {
        return super.buildStringHelper()
            .add("zipcode", zipcode)
            .add("country", country)
            .add("city", city)
            .add("streetName", streetName)
            .add("streetNo", streetNo)
    }
}